src/utils/TokenUtils.ts

```diff
@@ -114,7 +114,7 @@ export function isStablecoin(tokenSymbol: string): boolean {
 export function getTokenInfo(l2TokenAddress: string, chainId: number, tokenMapping = TOKEN_SYMBOLS_MAP): TokenInfo {
   const parsedAddress = chainIsSvm(chainId)
     ? toAddressType(l2TokenAddress).toBase58()
-    : toAddressType(l2TokenAddress).toEvmAddress();
+    : toAddressType(l2TokenAddress).__unsafeStaticCastToEvmAddress().formatAsChecksummedEvmAddress();
```

Comment:
I think this is a good use case for `toAddressType(l2TokenAddress, chainId).toNative()` (...or whatever the "to native representation" method is named. Then we can drop the local chainId evaluation.

(nb. this is however a bit of a weird place to use this, since `l2TokenAddress` is already a string, and we're just converting it back to a string. I feel like this function should probably accept an Address type instead of a string.)